Ui-Improvements

na50r opened the pull request:
In this branch I...

- Added a Typewriter that displays the time left more pleasantly than before.
- Added detailed explanation when a player clicks on Gamemode, tells them in more detail what exactly they're supposed to do. I did not remove the previous description because I'm against bombing players with descriptions. Descriptions should be put there when needed.
- I fixed up a bug on the result page, where profile pictures of users were not rendered correctly, I had to add something in on backend for [this](https://github.com/sopra-fs24-group-41/sopra-fs24-group-41-server/pull/199) **(This is up for debate)**
- I fixed a bug where the players were stuck in the result page even though the Lobby was closed
- I fixed a bug where the players were stuck in the result page even though the Lobby owner started a new game.
- I added a basic spinner animation (loading screen) when the API request takes longer than 1second
- I fixed a bug involving the timer. Player is forced to set timer each time a new game starts. Alternative is more painful to implement with reverse modulo operations and stuff. And there is also the danger that it fails and the wrong timer is picked, so fixed default timer is more robust. 
- I replaced the Player names with Player icons **(This is up for debate)**
- Added 'wiggle' animation to make @Tmmn happy 😉

Grizzlytron commented:
Good Work! From my side everything works.
Question:
- Is it intended that the wiggle also occurs, when a client clicks on the icon for the popup? Does not matter if logged in or not.
I understood that it would only wiggle, when the client is not logged in.

I took the liberty to add the names to the player list, because i thought it made more sense. Otherwise fine from my side

na50r commented:
> Question: Is it intended that the wiggle also occurs, when a client clicks on the icon for the popup? Does not matter if logged in or not. I understood that it would only wiggle, when the client is not logged in. Up to you

That part is still work in progress. 

> I took the liberty to add the names to the player list, because i thought it made more sense. Otherwise fine from my side

That's fine, thanks

na50r commented:
Update:
I fixed up the wordboard and added a search bar as requested on Discord.
![image](https://github.com/sopra-fs24-group-41/sopra-fs24-group-41-client/assets/145561869/6e1ed295-460a-42d3-8f2b-1e5ce95817b5)
